aws/src/main/java/org/apache/iceberg/aws/AwsProperties.java

```diff
@@ -1061,18 +1103,30 @@ public Map<String, String> httpClientProperties() {
     return httpClientProperties;
   }
 
+  public String clientRegion() {
+    return clientRegion;
+  }
+
+  public void setClientRegion(String clientRegion) {
+    this.clientRegion = clientRegion;
+  }
+
   /**
    * Configure the credentials for an S3 client.
    *
    * <p>Sample usage:
    *
    * <pre>
-   *     S3Client.builder().applyMutation(awsProperties::applyS3CredentialConfigurations)
+   *     S3Client.builder().applyMutation(awsProperties::applyCredentialConfigurations)
    * </pre>
    */
-  public <T extends S3ClientBuilder> void applyS3CredentialConfigurations(T builder) {
-    builder.credentialsProvider(
-        credentialsProvider(s3AccessKeyId, s3SecretAccessKey, s3SessionToken));
+  public <T extends AwsClientBuilder> void applyCredentialConfigurations(T builder) {
+    if (!Strings.isNullOrEmpty(this.clientCredentialsProvider)) {
+      builder.credentialsProvider(credentialsProvider(this.clientCredentialsProvider));
+    } else {
+      builder.credentialsProvider(
```

Comment:
also for `applyS3CredentialConfigurations`, I think if both `clientCredentialsProvider ` and `s3AccessKeyId, s3SecretAccessKey, s3SessionToken` configurations are set, we probably prefers to take the S3 one because it is an S3 level override.

Reply:
hi @jackye1995 I made the change in this method as u suggested. if s3 keys set, it will be used else credentials provider used if set. nothing set default provider will be used

Reply:
Sorry maybe I did not express clearly, what I meant was if both `clientCredentialsProvider`  and `s3AccessKeyId, s3SecretAccessKey, s3SessionToken` configurations are set, we should first see if `s3AccessKeyId, s3SecretAccessKey, s3SessionToken` is set, and use it if so, and otherwise check if `clientCredentialsProvider` is set and try use that

Reply:
hi @jackye1995 .  Sorry, I am still not getting. This is how I have implemented.

DefaultAwsClientFactory.s3() method calls applyS3CredentialConfigurations() method
```
public S3Client s3() {
      return S3Client.builder()
          .applyMutation(awsProperties::applyClientRegionConfiguration)
          .applyMutation(awsProperties::applyHttpClientConfigurations)
          .applyMutation(awsProperties::applyS3EndpointConfigurations)
          .applyMutation(awsProperties::applyS3ServiceConfigurations)
          .applyMutation(awsProperties::applyS3CredentialConfigurations)
          .applyMutation(awsProperties::applyS3SignerConfiguration)
          .build();
    }
```

then applyS3CredentialConfigurations method in AwsProperties calls credentialsProvider() method

```
public <T extends AwsClientBuilder> void applyS3CredentialConfigurations(T builder) {
    builder.credentialsProvider(
        credentialsProvider(s3AccessKeyId, s3SecretAccessKey, s3SessionToken));
  }
```

I have added this code to check credentialsProvider as fallback if s3 keys not set
```
  private AwsCredentialsProvider credentialsProvider(
      String accessKeyId, String secretAccessKey, String sessionToken) {
    if (accessKeyId != null) {
      if (sessionToken == null) {
        return StaticCredentialsProvider.create(
            AwsBasicCredentials.create(accessKeyId, secretAccessKey));
      } else {
        return StaticCredentialsProvider.create(
            AwsSessionCredentials.create(accessKeyId, secretAccessKey, sessionToken));
      }
    }

    if (!Strings.isNullOrEmpty(this.clientCredentialsProvider)) {
      return credentialsProvider(this.clientCredentialsProvider);
    }

    return DefaultCredentialsProvider.create();
  }
```
This is only for S3. All other clients will use `.applyMutation(awsProperties::applyCredentialConfigurations)`
```
  public <T extends AwsClientBuilder> void applyCredentialConfigurations(T builder) {
    if (!Strings.isNullOrEmpty(this.clientCredentialsProvider)) {
      builder.credentialsProvider(credentialsProvider(this.clientCredentialsProvider));
    }
  }
```

Reply:
oh I see did not see the line 

```
if (!Strings.isNullOrEmpty(this.clientCredentialsProvider)) {
      return credentialsProvider(this.clientCredentialsProvider);
    }
```

then I am good with this.